doseCalc/+DoseEngines/matRad_ParticlePencilBeamEngineAbstract.m

```diff
@@ -0,0 +1,902 @@
+                matRad_cfg.dispWarning('Machine data does not contain BAMStoIsoDist. Using default value of %f mm.',this.machine.meta.BAMStoIsoDist);
+            end
+            
+            %Biology
+            if ~isnan(this.constantRBE)
+                dij.RBE = this.constantRBE;
+            end
+            
+            % TODO: this is clumsy and needs to be changed with the
+            % biomodel update
+            if this.bioParam.bioOpt
+                this.calcBioDose = true;
+            end
+
+            % Load biologicla base data if needed
+            if this.calcBioDose
+                dij = this.loadBiologicalBaseData(cst,dij);
+                % allocate alpha and beta dose container and sparse matrices in the dij struct,
+                % for more informations see corresponding method
+                dij = this.allocateBioDoseContainer(dij);
+            end           
+
+            % allocate LET containner and let sparse matrix in dij struct
+            if this.calcLET
+                if isfield(this.machine.data,'LET')
+                    dij = this.allocateLETContainer(dij);
+                else
+                    matRad_cfg = MatRad_Config.instance();
+                    matRad_cfg.dispWarning('LET not available and will not be computed!');
+                    this.calcLET = false;
+                end
+            end
+
+            % lateral cutoff for raytracing and geo calculations
+            this.effectiveLateralCutOff = this.geometricLateralCutOff;
+        end
+
+        function currBeam = initBeam(this,dij,ct,cst,stf,i)
+            currBeam = initBeam@DoseEngines.matRad_PencilBeamEngineAbstract(this,dij,ct,cst,stf,i);
+            
+            %currBeam.rot_coordsVdoseGrid = currBeam.rot_coordsVdoseGrid(~isnan(currBeam.radDepthVdoseGrid{1}),:);
+            maxEnergy = max([currBeam.ray.energy]);
+            maxEnergyIx = find(this.round2(maxEnergy,4) == this.round2([this.machine.data.energy],4));
+            raShis = [currBeam.ray.rangeShifter];
+            minRaShi = min([raShis.eqThickness]);
+            radDepthOffset = this.machine.data(maxEnergyIx).offset + minRaShi;
+
+            % apply limit in depth
+            subSelectIx = currBeam.radDepths{1} < (this.machine.data(maxEnergyIx).depths(end) - radDepthOffset);
+            currBeam.ixRadDepths = currBeam.ixRadDepths(subSelectIx);
+            currBeam.subIxVdoseGrid = currBeam.subIxVdoseGrid(subSelectIx);
+            currBeam.radDepths = cellfun(@(rd) rd(subSelectIx),currBeam.radDepths,'UniformOutput',false);
+            currBeam.bevCoords = currBeam.bevCoords(subSelectIx,:);
+
+            %Precompute CutOff
+            this.calcLateralParticleCutOff(this.dosimetricLateralCutOff,currBeam);
+        end
+        
+        function dij = loadBiologicalBaseData(this,cst,dij)
+            matRad_cfg = MatRad_Config.instance();
+            if isfield(this.machine.data,'alphaX') && isfield(this.machine.data,'betaX')
```

Comment:
alphaX and betaX should be in the machine only for the carbon LEM model. For proton LET based biomodels these fields cannod be ssaved in machine.

Reply:
This should now also be resolved as biological dose calculation has been udpated.